src/key_store/mod.rs

```diff
@@ -0,0 +1,88 @@
+use std::collections::HashMap;
+
+use crate::{
+    ciphersuite::{CiphersuiteName, SignaturePublicKey, SignatureScheme},
+    credentials::{Credential, CredentialBundle, CredentialType},
+    extensions::Extension,
+    key_packages::{KeyPackage, KeyPackageBundle},
+};
+
+pub mod errors;
+
+pub use errors::KeyStoreError;
+
+#[derive(Debug, Default)]
+/// The `KeyStore` contains private key material of `Credentials` and
+/// `KeyPackage`s.
+pub struct KeyStore {
+    // Map from signature public keys to credential bundles
+    credentials: HashMap<SignaturePublicKey, CredentialBundle>,
+    key_packages: HashMap<Vec<u8>, KeyPackageBundle>,
+}
+
+impl KeyStore {
+    /// Retrieve a `KeyPackageBundle` from the key store given the hash of the
+    /// corresponding `KeyPackage`. This removes the `KeyPackageBundle` from the
+    /// store. Returns an error if no `KeyPackageBundle` can be found
+    /// corresponding to the given `KeyPackage` hash. TODO: This is not in use
+    /// yet, because the groups are not yet refactored to use the key store for
+    /// KeyPackageBundles.
+    pub(crate) fn _get_key_package_bundle(
```

Comment:
This came from https://github.com/openmls/openmls/pull/334#discussion_r570307673. The idea is that the key package is used in the creation of a group, after which it lives on in the group state. This will change eventually with the evolution of the key store (see above or #337).